SIGSEGV when running `helm lint` on traefik chart

### What happened?

Panic when linting traefik helm chart with Helm v4.0.0 but when using Helm v3, it works as expected.

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1fda40f]

goroutine 1 [running]:
helm.sh/helm/v4/pkg/chart/v2/lint/rules.validateCrdAPIVersion(0xc000b338f0?)
        helm.sh/helm/v4/pkg/chart/v2/lint/rules/crds.go:102 +0xf
helm.sh/helm/v4/pkg/chart/v2/lint/rules.Crds(0xc000b32270)
        helm.sh/helm/v4/pkg/chart/v2/lint/rules/crds.go:83 +0x765
helm.sh/helm/v4/pkg/chart/v2/lint.RunAll({0x7ffe337a3356?, 0xa?}, 0xc000b321b0, {0xc000b2cbd0, 0x7}, {0xc0002d9888, 0x2, 0x0?})
        helm.sh/helm/v4/pkg/chart/v2/lint/lint.go:68 +0x1e5
helm.sh/helm/v4/pkg/action.lintChart({0x7ffe337a3356, 0x1}, 0xc000b321b0, {0xc000b2cbd0, 0x7}, 0x0, 0x0)
        helm.sh/helm/v4/pkg/action/lint.go:129 +0x776
helm.sh/helm/v4/pkg/action.(*Lint).Run(0xc0003e7710, {0xc0002d9ba0, 0x1, 0x2?}, 0xc000b321b0)
        helm.sh/helm/v4/pkg/action/lint.go:63 +0xe5
helm.sh/helm/v4/pkg/cmd.newLintCmd.func1(0xc00011b300?, {0xc0009f4300?, 0x4?, 0x2763033?})
        helm.sh/helm/v4/pkg/cmd/lint.go:93 +0x2bf
github.com/spf13/cobra.(*Command).execute(0xc000978c08, {0xc0009f42d0, 0x1, 0x1})
        github.com/spf13/cobra@v1.10.1/command.go:1015 +0xb02
github.com/spf13/cobra.(*Command).ExecuteC(0xc000432308)
        github.com/spf13/cobra@v1.10.1/command.go:1148 +0x465
github.com/spf13/cobra.(*Command).Execute(0x2caa700?)
        github.com/spf13/cobra@v1.10.1/command.go:1071 +0x13
main.main()
        helm.sh/helm/v4/cmd/helm/helm.go:43 +0x159
```

Ref: https://github.com/traefik/traefik-helm-chart/issues/1568

### What did you expect to happen?

A valid result when running helm lint for the traefik chart like
```
helm lint .
==> Linting .

1 chart(s) linted, 0 chart(s) failed
```

### How can we reproduce it (as minimally and precisely as possible)?

1. git clone https://github.com/traefik/traefik-helm-chart.git
2. cd traefik-helm-chart/traefik
3. helm lint .

### Helm version

```console
version.BuildInfo{Version:"v4.0.0", GitCommit:"99cd1964357c793351be481d55abbe21c6b2f4ec", GitTreeState:"clean", GoVersion:"go1.25.3", KubeClientVersion:"v1.34"}
```


### Kubernetes version

```console
Client Version: v1.34.2
Kustomize Version: v5.7.1
```

--- a/internal/chart/v3/lint/rules/crds.go
+++ b/internal/chart/v3/lint/rules/crds.go
@@ -80,8 +80,10 @@ func Crds(linter *support.Linter) {
 				return
 			}
 
-			linter.RunLinterRule(support.ErrorSev, fpath, validateCrdAPIVersion(yamlStruct))
-			linter.RunLinterRule(support.ErrorSev, fpath, validateCrdKind(yamlStruct))
+			if yamlStruct != nil {
+				linter.RunLinterRule(support.ErrorSev, fpath, validateCrdAPIVersion(yamlStruct))
+				linter.RunLinterRule(support.ErrorSev, fpath, validateCrdKind(yamlStruct))
+			}
 		}
 	}
 }

--- a/internal/chart/v3/lint/rules/crds_test.go
+++ b/internal/chart/v3/lint/rules/crds_test.go
@@ -17,6 +17,8 @@ limitations under the License.
 package rules
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -33,4 +35,32 @@ func TestInvalidCrdsDir(t *testing.T) {
 
 	assert.Len(t, res, 1)
 	assert.ErrorContains(t, res[0].Err, "not a directory")
+}
+
+// multi-document YAML with empty documents would panic
+func TestCrdWithEmptyDocument(t *testing.T) {
+	chartDir := t.TempDir()
+
+	os.WriteFile(filepath.Join(chartDir, "Chart.yaml"), []byte(
+		`apiVersion: v1
+name: test
+version: 0.1.0
+`), 0644)
+
+	// CRD with comments before --- (creates empty document)
+	crdsDir := filepath.Join(chartDir, "crds")
+	os.Mkdir(crdsDir, 0755)
+	os.WriteFile(filepath.Join(crdsDir, "test.yaml"), []byte(
+		`# Comments create empty document
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: test.example.io
+`), 0644)
+
+	linter := support.Linter{ChartDir: chartDir}
+	Crds(&linter)
+
+	assert.Len(t, linter.Messages, 0)
 }

--- a/pkg/chart/v2/lint/rules/crds.go
+++ b/pkg/chart/v2/lint/rules/crds.go
@@ -80,8 +80,10 @@ func Crds(linter *support.Linter) {
 				return
 			}
 
-			linter.RunLinterRule(support.ErrorSev, fpath, validateCrdAPIVersion(yamlStruct))
-			linter.RunLinterRule(support.ErrorSev, fpath, validateCrdKind(yamlStruct))
+			if yamlStruct != nil {
+				linter.RunLinterRule(support.ErrorSev, fpath, validateCrdAPIVersion(yamlStruct))
+				linter.RunLinterRule(support.ErrorSev, fpath, validateCrdKind(yamlStruct))
+			}
 		}
 	}
 }

--- a/pkg/chart/v2/lint/rules/crds_test.go
+++ b/pkg/chart/v2/lint/rules/crds_test.go
@@ -17,6 +17,8 @@ limitations under the License.
 package rules
 
 import (
+	"os"
+	"path/filepath"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
@@ -33,4 +35,32 @@ func TestInvalidCrdsDir(t *testing.T) {
 
 	assert.Len(t, res, 1)
 	assert.ErrorContains(t, res[0].Err, "not a directory")
+}
+
+// multi-document YAML with empty documents would panic
+func TestCrdWithEmptyDocument(t *testing.T) {
+	chartDir := t.TempDir()
+
+	os.WriteFile(filepath.Join(chartDir, "Chart.yaml"), []byte(
+		`apiVersion: v1
+name: test
+version: 0.1.0
+`), 0644)
+
+	// CRD with comments before --- (creates empty document)
+	crdsDir := filepath.Join(chartDir, "crds")
+	os.Mkdir(crdsDir, 0755)
+	os.WriteFile(filepath.Join(crdsDir, "test.yaml"), []byte(
+		`# Comments create empty document
+---
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: test.example.io
+`), 0644)
+
+	linter := support.Linter{ChartDir: chartDir}
+	Crds(&linter)
+
+	assert.Len(t, linter.Messages, 0)
 }

